subsys/bluetooth/host/att.c

```diff
@@ -149,6 +149,13 @@ static int chan_send(struct bt_att_chan *chan, struct net_buf *buf,
 
 	BT_DBG("code 0x%02x", hdr->code);
 
+	if (hdr->code == BT_ATT_OP_MTU_REQ && chan->chan.tx.cid != BT_L2CAP_CID_ATT) {
```

Comment:
This check only makes sense if we support EATT: 

```suggestion
	if (IS_ENABLED(CONFIG_BT_EATT) && 
	    hdr->code == BT_ATT_OP_MTU_REQ && chan->chan.tx.cid != BT_L2CAP_CID_ATT) {
```

Reply:
The way I am reading the spec it is not allowed to do MTU exchange on dynamically allocated unenhanced channels either.

> Vol 3, Part G 4.2
The Exchange MTU sub-procedure shall only be supported on the LE Fixed Channel Unenhanced ATT bearer.

I assume it is possible to set up additional unenhanced ATT bearers using dynamically allocated L2CAP channels?

Reply:
> The way I am reading the spec it is not allowed to do MTU exchange on dynamically allocated unenhanced channels either.
> 
> > Vol 3, Part G 4.2
> > The Exchange MTU sub-procedure shall only be supported on the LE Fixed Channel Unenhanced ATT bearer.
> 
> I assume it is possible to set up additional unenhanced ATT bearers using dynamically allocated L2CAP channels?

Not as far as I know. If I remember correctly there can only be a single unenhanced ATT bearer, and that is on a fixed L2CAP channel.

Reply:
> The L2CAP channel mode determines the behavior of Attribute protocol on that ATT bearer. If the L2CAP channel mode is using Enhanced Credit Based Flow Control Mode, the ATT bearer is known as an Enhanced ATT bearer. Any ATT bearer that is not an Enhanced ATT bearer, using any other L2CAP channel mode, is known as an Unenhanced ATT bearer.

> When using an Unenhanced ATT bearer, the channel used to carry Attribute protocol PDUs over LE is the Attribute L2CAP fixed channel.

> To establish an Unenhanced ATT bearer, the Channel Establishment procedure (as defined in [Vol 3] Part C, Section 7.2) shall be used with the PSM set to ATT.